### PR TITLE
libarchive: fix patch

### PR DESCRIPTION
**Pull request by Cpasjuste**
libarchive was compiling fine but couldn't link (undefined reference to waitpid and such). 
Tested working on my nestopia (pnes) port.

### CODE REVIEW

File: libarchive/libarchive-3.3.2-vita.patch [hunk 1]
```diff
@@ -22,9 +22,32 @@ diff -Naur libarchive-3.3.2.orig/libarchive/archive_read_disk_posix.c libarchive
...
  #define O_BINARY	0
+diff -Naur libarchive-3.3.2.orig/libarchive/archive_read_support_filter_program.c libarchive-3.3.2/libarchive/archive_read_support_filter_program.c
+--- libarchive-3.3.2.orig/libarchive/archive_read_support_filter_program.c	2017-07-10 04:38:04.000000000 +0200
++++ libarchive-3.3.2/libarchive/archive_read_support_filter_program.c	2022-05-12 12:27:32.993098967 +0200
+@@ -58,7 +58,13 @@
+ #include "archive_read_private.h"
+ #include "filter_fork.h"
+ 
+-
++#if 1
```
Comment (isage) [inline libarchive/libarchive-3.3.2-vita.patch:33]:
```suggestion
+#ifdef __vita__
```

Comment (Cpasjuste) [reply libarchive/libarchive-3.3.2-vita.patch:33]: Hi, i'm actually porting this to multiple platforms, it's the reason of this. Is it really useful to add a  specific define here ?

Comment (isage) [reply libarchive/libarchive-3.3.2-vita.patch:33]: Well, if you ever want to upstream those changes - yes

Comment (Cpasjuste) [reply libarchive/libarchive-3.3.2-vita.patch:33]: I have to redo the whole patch then,  because it's full of this :)

**Comment by Cpasjuste** [comment]
Hi @isage , I think the PR is clean now, can you merge it (I need it to update my workflows)? Thanks!

**Comment by isage** [comment]
I don't have rights

**Comment by Cpasjuste** [comment]
Thanks mate :)

**Pull request merged**